Sebastian-Sole/handicappin
# ✨ feat: dashboard

## What's new?
- A dashboard which displays the user's handicap, and past rounds in a graph and detailed table.
- DB migrations which redefine Round and Course tables to facilitate easy display of data/statistics
- Splitting types for Round to be FormRound and RoundMutation for filling out the form and sending the data to the DB
- Charts, Tables, Skeleton components
- Added isMounted to deal with hydration issues
- Corrected calculations for handicap and scoring to accurately handle 9 and 18 holes of golf
- Replaced authButtons with correct way of implementing navigational buttons

## How to test
- Visit [prod](https://vercel.live/link/handicappin-git-feat-dashboard-sebastiansoles-projects.vercel.app?via=deployment-domains-list-branch)
- Create user (signing up with Github/Google is not implemented)
- Add a round (click "Populate DB" to add dummy data)
- Visit the dashboard by clicking the button in the navbar
- Verify the rounds have been added in, the handicap has been calculated, and the table functionality works correctly
- Check responsivity and that a user cannot access a different dashboard than their own

## Future Work
- Add pages for the links on dashboard (wip)
